include/OsqpEigen/Data.tpp

```diff
@@ -32,7 +32,9 @@ bool OsqpEigen::Data::setHessianMatrix(const Eigen::SparseMatrix<T> &hessianMatr
     }
 
     //set the hessian matrix
-    if(!OsqpEigen::SparseMatrixHelper::createOsqpSparseMatrix(hessianMatrix, m_data->P)){
+    // osqp 0.6.0 required only the upper triangular part of the hessian matrix
+    Eigen::SparseMatrix<T> hessianMatrixUpperTriangular = hessianMatrix.template triangularView<Eigen::Upper>();
```

Comment:
Is this method allocating memory?

Reply:
Good point. Could we move this to be a buffer in the class, or we can't because the method is templated?

Reply:
I think it allocates memory

Reply:
I would suggest to keep a copy of the hessian and iterate over the upper triangular non-zeros of the input hessian.

Reply:
This function is called only the first time. 
Probably there is a way to avoid to allocate memory

Reply:
In which sense "only the first time"? It is not called every time the hessian changes?

Reply:
No, the function called every time the hessian changes is 
https://github.com/robotology/osqp-eigen/blob/bbd96eaaa9cb6fdb4f3a7272817e4f832186ce5b/include/OsqpEigen/Solver.tpp#L13

Reply:
Ah, sorry. Then it makes sense.

Reply:
I thought as well it was called every time. So there are no problems I guess.

Reply:
Unfortunately, it is not easy to extract the upper triangular matrix from a CSC representation (representation used by osqp).  Furthermore, the methods used to retrieve the `outerIndexPtr`  and the `innerIndexPtr` are defined in  [`Eigen::SparseCompressedBase`](https://eigen.tuxfamily.org/dox/classEigen_1_1SparseCompressedBase.html) and the 
[`Eigen::TriangularView`](https://eigen.tuxfamily.org/dox/classEigen_1_1TriangularView.html) class does not inherit from `Eigen::SparseCompressedBase`. 

This is the reason why the following line is needed
```cpp
Eigen::SparseMatrix<T> hessianMatrixUpperTriangular = hessianMatrix.template triangularView<Eigen::Upper>();
```